third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1865,17 +1998,31 @@ impl ::std::fmt::Debug for Bytecode {
             Bytecode::CallGeneric(a) => write!(f, "CallGeneric({})", a),
             Bytecode::Pack(a) => write!(f, "Pack({})", a),
             Bytecode::PackGeneric(a) => write!(f, "PackGeneric({})", a),
+            Bytecode::PackVariant(a) => write!(f, "PackVariant({})", a),
+            Bytecode::TestVariant(a) => write!(f, "TestVariant({})", a),
+            Bytecode::PackVariantGeneric(a) => write!(f, "PackVariantGeneric({})", a),
+            Bytecode::TestVariantGeneric(a) => write!(f, "TestVariantGeneric({})", a),
```

Comment:
I highly doubt if it is still a good idea to have generic and non-generic versions of the same instructions. 

Back in the old days, the split was introduced in hoping that this would make the bytecode more optimized for the VM to execute. However now it's clear this point is totally moot.
- There is a strong need for the bytecode format to be a stable, backward compatible storage format, which conflicts with being optimized for execution
  - A more suitable solution to this problem is to introduce some VM-internal microcodes that are designed to be unstable, adapting to VM changes to the VM's execution engine. Notice that this is exactly how modern CPUs architectures work.
- Because of the two versions, we ended up having to duplicate a ton of code. This has been a maintenance nightmare and greatly increased the chance we screw something up (e.g. making changes in only one version)

Reply:
This PR is no redesigning the bytecode architecture, but following along what we have. One principle of the current design is that almost every instruction has exactly one argument, therefore fits into three bytes in most cases (one for the bytecode, one for the argument, one terminator). This is not for efficiency of execution but for the size of the bytecode. (You are right that for performance of execution it makes little sense.) The entire way how the file format is designed is for this.

If we want to change this, we can't do it just for two out of dozens instructions. The resulting design would look very, very odd.  So we have BorrowField and BorrowFieldGeneric, and symmetrically, we have BorrowVariantField and BorrowVariantFieldGeneric. I don't think that any other approach leads to a satisfactory design.

Reply:
I see your point. Agreed that we could try to come up with a more appropriate solution to this problem later.